src/component/ListContents.js

```diff
@@ -1,28 +1,81 @@
 import React from 'react';
 import { withFirestore } from 'react-firestore';
 import { Link } from 'react-router-dom';
+import dayjs from 'dayjs';
+import calculateNewPurchaseValues from '../calculations';
+
+const ListContents = ({ listData, firestore, token }) => {
+  const today = new Date();
+  const items = listData.items;
+  const itemsDocRef = firestore
+    .collection('lists')
+    .doc(token)
+    .collection('items');
+
+  // function to change database on button click
+  const handlePurchase = event => {
+    let itemId = event.target.id;
+    event.preventDefault();
+    itemsDocRef
+      .doc(itemId)
+      .get()
+      .then(doc => {
+        return calculateNewPurchaseValues(doc.data(), today);
+      })
+      .then(updateDatabase);
+  };
+
+  const updateDatabase = data => {
+    itemsDocRef.doc(data.id).update({
+      numberOfDays: data.numberOfDays,
+      dateOfPurchase: data.dateOfPurchase,
+      numberOfPurchases: data.numberOfPurchases,
+    });
+  };
+
+  // This function is called from within className prop
+  // each time items get rendered & it sets the class
+  // based on whether an item has been purchased within
+  // the last 24 hours
+  const calculateIfPurchased = item => {
+    // returns true if item was purchased within last 24 hours
+    const wasItemPurchasedToday = today => {
+      today = dayjs(today);
+      return today.diff(dayjs(item.dateOfPurchase.toDate()), 'hour') <= 24;
+    };
+
+    if (item.dateOfPurchase === undefined) {
+      return 'nonPurchasedItem';
+    } else if (wasItemPurchasedToday(today)) {
+      return 'purchasedItem';
+    } else {
+      return 'nonPurchasedItem';
+    }
+  };
```

Comment:
**No need to change,** just an observation for the future:

Since `undefined` is a `falsy` value, you can omit the triple `=` and save some keystrokes. So this would become:

```js
if (!item.dateOfPurchase)
```

You can read more about falsy values here: https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Another observation:

It's usually a good practice to store _almost_ all your strings in constants, picture this:

1. You return `nonPurchasedItem`
2. In other place, you check for `if ( x === 'nonPuchasedItem')`
3. The code inside the if statement will never be executed, because I missed the `r` in `nonPuRchased`

If you use a constant instead of a plain string, you will always check on the constant, and you can't misspelled that b/c the linter will catch that error, or the program just won't compile.

```js
// constants.js
export const NON_PURCHASED_ITEM = 'nonPurchasedItem';

// ListContents.js

import { NON_PURCHASED_ITEM } from './constants'; // the file above

// later on...
if (!item.dateOfPurchase) {
  return NON_PURCHASED_ITEM;
}
```